drivers/SmartThings/zigbee-window-treatment/src/IKEA/init.lua

```diff
@@ -21,44 +21,50 @@ local WindowCovering = zcl_clusters.WindowCovering
 local SHADE_SET_STATUS = "shade_set_status"
 
 local function current_position_attr_handler(driver, device, value, zb_rx)
-  local level = 100 - value.value
-  local current_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID, capabilities.windowShadeLevel.shadeLevel.NAME)
-  local windowShade = capabilities.windowShade.windowShade
-  if level == 0 then
-    device:emit_event(windowShade.closed())
-    device:emit_event(capabilities.windowShadeLevel.shadeLevel(0))
-  elseif level == 100 then
-    device:emit_event(windowShade.open())
+  if value.value == 255 then
```

Comment:
can you make this check line up with the other `level == ` checks, perhaps `level == -155`?

Reply:
The Current Position Lift Percentage attribute is of type uint8, so a check against a negative number is not possible.

Reply:
I _believe_ it gets cast to a regular old lua `number` in the driver code, but a test should be able to confirm

Reply:
The value reported via zigbee is 0xFF if the position is unkown according to the zigbee cluster library. If smartthings does not change this value we would like to check against the standard value.

Reply:
if you'd left the code as is, with the first line `local level = 100 - value.value`, I would expect that, if `value.value` is 0xFF, that after that line executes, `level` will have the value of `-155`. Keeping the structure the same, of checking values of `level` would make the code more readable, rather than having a breakout section to test specifically for `0xFF` and putting all other existing checks behind an `else`

Reply:
Now I understand. Indeed nesting the code like it was isn't great so I imporved the code now.